lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU

```diff
@@ -155,7 +164,7 @@ ffMaint(
 
 // Trip the beam for unknown state
 ffUnknown(
-    i_xOK := getState <= 0 AND NOT bInTransition,
+    i_xOK := getState > 0 OR bInTransition,
```

Comment:
Friday afternoon is making logic harder than it should be. Would you mind explaining the logic inversion change here to put my confusion to rest?

`getState > 0` means in known state
`bInTransition` means arbiter beam parameter transition in progress

```
OK = getState > 0 OR bIntransition
Not OK = not (getState > 0 OR bIntransition)
Not OK = getState <= 0 AND NOT bIntransition
```

Not OK if in an unknown state _specifically when not in a BPTM transition_
OK if in a known state or in a BTPM transition

Reply:
This will be a bit rambley

In the PR description I should have put: "fixed the inverted logic for the unknown state FFO" instead of "Add a fast fault for the case where we've stopped moving between states.", oops.

The logic from last PR was exactly inverted. I forgot that the API was "ok => no fault" and I had tried to put in "fault if state is unknown and not doing a bptm transition", instead the pre-diff code reads "things are ok only if the state is unknown and we are not doing a bptm transition". As a result, we had a permenant fast fault when at known states.

Your description of the current behavior is correct: I want a fault in the cases where our state is unknown. That is, if we start up the PLC and a device is already between states, that's a fault, and if a device stops between states, that's also a fault.

Previously, this was taken care of by putting in an "Unknown" state assertion to the arbiter at "no beam".
Now, with database-generated states, this would require an additional "Unknown" database entry for every device, since we need to have unique IDs for the arbiter to know that it's a different set of beam parameters. So, rather than put the burden on the database, do the check here: enforce no beam via fast fault.

Reply:
And further, I don't want a fault in transition, I want to retain the option for a "transition state" to have nonzero beam parameters for cases where that's OK.

Reply:
Thanks for the explanation - that does clear it all up 👍

Reply:
Thank you for making me explain/justify/document the reasoning, it was definitely missing